src/Knp/DictionaryBundle/Dictionary/Factory/FactoryAggregate.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Knp\DictionaryBundle\Dictionary\Factory;
+
+use Knp\DictionaryBundle\Dictionary\Factory;
+
+class FactoryAggregate implements Factory
+{
+    /**
+     * @var Factory[]
+     */
+    private $factories = array();
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create($name, array $config)
+    {
+        foreach ($this->factories as $factory) {
+            if ($factory->supports($config)) {
+                return $factory->create($name, $config);
+            }
+        }
+
+        throw new \InvalidArgumentException(sprintf(
+            'The dictionary with named "%s" cannot be created.',
+            $name
+        ));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports(array $config)
+    {
+        return true;
+    }
+
+    /**
+     * @param Factory $factory
+     *
+     * @return FactoryAggregate
+     */
+    public function addFactory(Factory $factory)
```

Comment:
I think should be the very first method of this class because it's the way to build the instance.